src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/alignment/AssemblyContigAlignmentsConfigPicker.java

```diff
@@ -520,92 +684,71 @@ static GoodAndBadMappings splitGaps(final GoodAndBadMappings configuration) {
         return new GoodAndBadMappings(good, bad, configuration.goodMappingToNonCanonicalChromosome);
     }
 
-    @VisibleForTesting
-    static boolean gappedAlignmentOffersBetterCoverage(final AlignmentInterval gapped,
-                                                       final AlignmentInterval overlappingNonGapped) {
-        final int diff = gapped.getSizeOnRead() - overlappingNonGapped.getSizeOnRead();
-        if ( diff == 0) {
-            return gapped.alnScore > overlappingNonGapped.alnScore;
-        } else {
-            return diff > 0;
-        }
-    }
-
-    //==================================================================================================================
-
     /**
-     * For contigs with more than 1 best-scored configurations as determined by
-     * {@link #pickBestConfigurations(AlignedContig, Set, Double)},
-     * save the contigs that has one and only one configuration that
-     * has all mapping quality strictly above the specified {@code threshold}.
-     * Example:
-     *  if a contig has two equal scored configurations with MQ's {10, 60, 60}, and {60, 60},
-     *  this function will favor/pick the {60, 60} configuration if the {@code threshold} is 10,
-     *  hence remove the ambiguity;
-     *  on the other hand if the {@code threshold} is passed in as any value below 10, say 0,
-     *  then this function returns all the original {@code differentRepresentationsForOneContig}
+     * See {@link #splitGaps(GoodAndBadMappings, boolean)}.
+     * This implementation drops alignment that are contained by others, in terms of their read span.
      */
     @VisibleForTesting
-    public static List<GoodAndBadMappings> filterSecondaryConfigurationsByMappingQualityThreshold(
-            final List<GoodAndBadMappings> differentRepresentationsForOneContig,
-            final int threshold) {
+    static GoodAndBadMappings splitGapsAndDropAlignmentContainedByOtherOnRead(final GoodAndBadMappings configuration) {
+        final List<AlignmentInterval> originalGoodMappings = configuration.goodMappings;
+        final List<Tuple2<Boolean, AlignmentInterval>> alignmentSplitChildren = originalGoodMappings.stream()
```

Comment:
It seems weird to put these in a `List<Tuple2<Boolean, AligmentInterval>>`. Why not first just make a `List<AlignmentInterval>`. Then loop through it with i and j indices as below, but instead of resetting the tuple in this list just pop them into the `good` and `bad` lists below?

Reply:
Yep. done